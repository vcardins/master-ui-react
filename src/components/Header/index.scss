@import '~variables';

#app[data-viewport="mini"],
#app[data-viewport="tiny"] {
	#header {
		.profile {
			.info {
				opacity: 0;
				display: none;
			}
		}
	}
}

.collapsed {
	#header {
		.header-left {
			margin-left: 0em;
		}
	}
}

.vertical-menu {
	.header-central {
		.menu {
			display: none;
		}
	}
}

#header {
	line-height: 40px;
	height: 40px;
	background:
	linear-gradient(to top, rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.25))
	var(--themeColor);

	a {
		color: var(--themeLightestColor);

		&:hover {
			color: $white;
		}
	}

	/* linear-gradient(to bottom, $color-medium, $color-darkest); */
	padding: 0;
	color: $white;

	.header-left,
	.header-central,
	.header-right {
		display: inline-block;
	}

	.header-left {
		width: $navbar-width-expanded - 50;
		font-weight: 400;
		line-height: normal;
		margin-left: 1.25em;
		font-size: 20px;
		text-align: center;
		transition: margin-left 0.2s;

		.header-logo {
			text-align: center;
			display: inherit;
		}

		.header-title {
			margin-left: 0.5em;
		}
	}

	.header-central {
		margin-left: 3em;

		a {
			line-height: normal;
		}
	}

	.header-right {
		float: right;
		margin-right: 1em;

		img.image {
			margin-right: 1em;
		}

		.dropdown {
			display: inline-flex;
		}

		.profile {
			display: inline-block;
			margin-right: 1em;
			vertical-align: middle;

			.photo,
			.info {
				display: inline-block;
			}

			.photo {
				vertical-align: top;

				.avatar {
					margin-right: 1em;
				}
			}

			.info {
				line-height: initial;
				vertical-align: inherit;
				opacity: 1;
				transition: opacity 0.5s ease;

				&>span {
					display: block;
				}

				.greetings {
					font-size: 75%;
					line-height: 10px;
				}
			}
		}

		.logout {
			display: inline-block;
			vertical-align: bottom;
			width: $navbar-width-collapsed - 20;
			text-align: center;
		}
	}
}
