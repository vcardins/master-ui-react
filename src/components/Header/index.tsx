import * as React from 'react';
import { browserHistory, Link } from 'react-router';
import { Icon, Image, Dropdown } from 'semantic-ui-react';
import Logo from '../Logo';
import IMenuItem from '../IMenuItem';
import { UserProfile } from 'core/auth';
import appSettings from 'core/settings';
import toast from 'core/helpers/toast';
import Nav from '../Nav';
import './index.scss';

interface Props {
    user?: any;
    title?: string;
    onLogout: () => void;    
    onTogglePanel: (side: string) => void;
    routes?: Array<IMenuItem>;
    activeRoute?: string;
}

const Header: React.StatelessComponent<Props> = (props: Props): JSX.Element => {
    const { title, onTogglePanel, onLogout, user = {}, routes, activeRoute } = props;

    const trigger = (
        <div className="profile">
            <div className="photo">
                <Image avatar src={`https://avatars.io/twitter/${user.username}`} /> 
            </div>
            <div className="info">
                <span className="greetings">Hello,</span> 
                <span className="name">{ user.firstName }</span>
            </div>
        </div>
    );

    const options = [
        { key: 'settings', text: 'Profile', icon: 'user', onClick: () => browserHistory.replace('/user-profile')  },
        { key: 'toast', text: 'Show Notification', icon: 'sign out', onClick: () => toast.info('Message') },
        { key: 'sign-out', text: 'Sign Out', icon: 'sign out', onClick: onLogout },
    ];

    const DropdownUserProfile = () => (
        <Dropdown trigger={trigger} options={options} pointing="top right" icon={null} />
    );

    return (
        <header id="header">
            <div className="header-left">
                <span className="header-logo">
                    <Icon name="diamond"/>
                </span>
                <span className="header-title">
                    <Link to="/">{ title }</Link>
                </span>
            </div>
            <div className="header-central">
                { routes && <Nav routes={routes} position="horizontal" activeRoute={activeRoute}/> }
            </div>
            <div className="header-right">                
                <DropdownUserProfile/>                
            </div>
        </header>
    );
};

export default Header;
