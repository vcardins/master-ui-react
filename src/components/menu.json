{
	"items": [
		{ 
			"id": "dashboard", 
			"href": "/", 
			"label": "Dashboard", 
			"icon": "dashboard" 
		},
		{ 
			"id": "geography", 
			"href": "", 
			"label": "Locations", 
			"icon": "map",
			"children": [ 
				{ "id": "countries", "href": "/countries", "label": "Countries", "icon": "flag" },
				{ "id": "city", "href": "/provinces", "label": "Provinces", "icon": "map" }
			]
		},
		{ 
			"id": "notebook", 
			"href": "/notebook", 
			"label": "Notebook", 
			"icon": "book" 
		},
		{ 
			"id": "reports", 
			"href": "/reports", 
			"label": "Reports", 
			"icon": "calculator" 
		}
	]
}
