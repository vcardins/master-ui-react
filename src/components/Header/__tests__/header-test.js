/* eslint-env node, jest, mocha, jsx */
import * as React from 'react';
import * as Renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from '../index';
import * as menu from '../../menu.json';

const user = {
  userId: 1,
  username: 'admin',
  firstName: 'System',
  lastName: 'Admin',
  displayName: 'System A.',
  fullName: 'System Admin',
  lastLogin: '2017-05-02T07:40:18.1538029+00:00',
  email: 'admin@master-ui.com',
  profilePhoto: null,
  twitterId: '@masterui',
  skypeId: 'masterui',
  whatsAppId: '00000000',
  facebookId: 'masterui',
  bio: '',
  iso2: 'US',
  avatar: '',
  coverPhoto: 'unknown.jpg',
  preferences: {
	showMyLocation: true,
	shareProfileAsPublic: true,
	allowNotifications: true,
	shareWhatsAppId: true,
	shareSkypeId: true,
	shareTwitterId: true,
	shareFacebookId: true,
	dateFormat: 'ddd, MMM d, yyyy',
	timezoneId: 56,
	utcOffset: 12.0,
	coverPhoto: 'unknown.jpg',
	avatar: 'unknownd',
  },
  status: {
	userId: 0,
	statusId: 1,
	availability: 'Online',
	message: null,
  },
};

const defaultProps = { 
	user, 
	title: 'Master UI',
	routes: menu['items'],
	activeRoute: '',
	onLogout: jest.fn(),
	onTogglePanel: jest.fn(),
};

function setup (props, test) {
	return () => test(
		Renderer.create(
			<Header {...props} />
		)
	);
}

describe('Header Component Rendering', () => {
	it('Should exist', () => {
		expect(Header).not.toBeNull();
	});

	it('Should render something if it receives required props', setup(
		defaultProps,
		(component) => expect(component.toJSON()).toMatchSnapshot()
	));	
});

describe('Header behaviour', () => {
	it('should pass a selected value to the onClick handler', setup(
		defaultProps,
		() => {
			const wrapper = shallow(<Header {...defaultProps} />);
			expect(wrapper).toMatchSnapshot();
		}
	));	
})