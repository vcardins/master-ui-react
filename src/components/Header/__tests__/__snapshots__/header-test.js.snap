// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Rendering Should render something if it receives required props 1`] = `
<header
  id="header"
>
  <div
    className="header-left"
  >
    <span
      className="header-logo"
    >
      <i
        aria-hidden="true"
        className="diamond icon"
      />
    </span>
    <span
      className="header-title"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        Master UI
      </a>
    </span>
  </div>
  <div
    className="header-central"
  >
    <ul
      className="menu horizontal"
    >
      <li
        className=""
        id="mnu-dashboard"
        onClick={[Function]}
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="dashboard icon"
          />
          Dashboard
        </a>
      </li>
      <li
        className="active parentLevel"
        id="mnu-geography"
        onClick={[Function]}
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="map icon"
          />
          Locations
          <i
            aria-hidden="true"
            className="icon chevron down expander"
          />
        </a>
        <ul
          className="submenu"
        >
          <li
            className=""
            id="mnu-countries"
            onClick={[Function]}
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Countries
            </a>
          </li>
          <li
            className=""
            id="mnu-city"
            onClick={[Function]}
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Provinces
            </a>
          </li>
        </ul>
      </li>
      <li
        className=""
        id="mnu-notebook"
        onClick={[Function]}
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="book icon"
          />
          Notebook
        </a>
      </li>
      <li
        className=""
        id="mnu-reports"
        onClick={[Function]}
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="calculator icon"
          />
          Reports
        </a>
      </li>
    </ul>
  </div>
  <div
    className="header-right"
  >
    <div
      aria-busy={undefined}
      aria-disabled={undefined}
      aria-expanded={false}
      aria-multiselectable={undefined}
      className="ui top right pointing dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        className="profile"
      >
        <div
          className="photo"
        >
          <img
            alt={undefined}
            className="ui avatar image"
            height={undefined}
            src="https://avatars.io/twitter/admin"
            width={undefined}
          />
        </div>
        <div
          className="info"
        >
          <span
            className="greetings"
          >
            Hello,
          </span>
          <span
            className="name"
          >
            System
          </span>
        </div>
      </div>
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <i
            aria-hidden="true"
            className="user icon"
          />
          <span
            className="text"
          >
            Profile
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <i
            aria-hidden="true"
            className="sign out icon"
          />
          <span
            className="text"
          >
            Show Notification
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <i
            aria-hidden="true"
            className="sign out icon"
          />
          <span
            className="text"
          >
            Sign Out
          </span>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header behaviour should pass a selected value to the onClick handler 1`] = `
<header
  id="header"
>
  <div
    className="header-left"
  >
    <span
      className="header-logo"
    >
      <Icon
        as="i"
        name="diamond"
      />
    </span>
    <span
      className="header-title"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/"
      >
        Master UI
      </Link>
    </span>
  </div>
  <div
    className="header-central"
  >
    <Nav
      activeRoute=""
      position="horizontal"
      routes={
        Array [
          Object {
            "href": "/",
            "icon": "dashboard",
            "id": "dashboard",
            "label": "Dashboard",
          },
          Object {
            "children": Array [
              Object {
                "href": "/countries",
                "icon": "flag",
                "id": "countries",
                "label": "Countries",
              },
              Object {
                "href": "/provinces",
                "icon": "map",
                "id": "city",
                "label": "Provinces",
              },
            ],
            "href": "",
            "icon": "map",
            "id": "geography",
            "label": "Locations",
          },
          Object {
            "href": "/notebook",
            "icon": "book",
            "id": "notebook",
            "label": "Notebook",
          },
          Object {
            "href": "/reports",
            "icon": "calculator",
            "id": "reports",
            "label": "Reports",
          },
        ]
      }
    />
  </div>
  <div
    className="header-right"
  >
    <DropdownUserProfile />
  </div>
</header>
`;
